src/System.Text.Json/tests/JsonReaderStateAndOptionsTests.cs

```diff
@@ -65,6 +65,30 @@ public static void JsonReaderStateCtor()
             Assert.Equal(expectedOption, state.Options);
         }
 
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(3)]
+        [InlineData(byte.MaxValue)]
+        [InlineData(byte.MaxValue + 4)] // Other values, like byte.MaxValue + 1 overflows to 0 (i.e. JsonCommentHandling.Disallow), which is valid.
+        public static void TestCommentHandlingInvalid(int enumValue)
+        {
+            try
+            {
+                var options = new JsonReaderOptions { CommentHandling = (JsonCommentHandling)enumValue };
+                Assert.True(false, "Expected ArgumentException was not thrown. CommentHandling enum must be valid.");
+            }
+            catch (ArgumentException)
+            { }
```

Comment:
This should use AssertExtensions.Throws, should validate the argument name in the exception, etc.  Same below.

Reply:
We don't really have an argument name. It's just "value".

Reply:
"value" is an argument name.